### PR TITLE
Expose Flutter detection to clients, with better naming

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: lib/src/sdk_env.dart [hunk 1]
```diff
@@ -185,15 +186,20 @@ class PubEnvironment {
     }
   }
 
-  Future<ProcessResult> runUpgrade(String packageDir, bool isFlutter,
+  Future<bool> detectFlutterUse(String packageDir) async {
```
Comment (jakobr-google) [inline lib/src/sdk_env.dart]: Does this need to be async?

Comment (isoos) [reply lib/src/sdk_env.dart]: Doesn't need to, but I want to remove the non-async parts eventually, and it is better to start with the async API.

Comment (isoos) [reply lib/src/sdk_env.dart]: * non-async parts = parts where we use sync `dart:io`, while we could have used async instead.

Comment (jakobr-google) [reply lib/src/sdk_env.dart]: sgtm

**Pull request merged**